include/PriorityQueue.h

```diff
@@ -9,8 +9,32 @@ class PriorityQueue {
 
 public:
     void push(const T &t);
+
     T poll();
 
 private:
+    struct {
+        T operator()(T a, T b) const {
```

Comment:
Notice that marking the arguments simply as T means they will be copied every call to the comparator. This is bad for performance and sometimes even impossible (i.e - if T is or contains a unique_ptr or any other type that deleted the Copy-constructor).
So in this case it definitely better to use "const-reference".
Check it out: https://www.geeksforgeeks.org/difference-between-call-by-value-and-call-by-reference/